src/main/java/spoon/MavenLauncher.java

```diff
@@ -0,0 +1,233 @@
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon;
+
+import org.apache.maven.model.Build;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Create a Spoon launcher from a maven pom file
+ */
+public class MavenLauncher extends Launcher {
+	private String m2RepositoryPath;
+
+	public MavenLauncher(String pomPatch, boolean includeTest) {
+		this(pomPatch, Paths.get(System.getProperty("user.home"), ".m2", "repository").toString(), includeTest);
+	}
+
+	/**
+	 *
+	 * @param projectRoot the path to the root of the project (the folder that contains the pom)
+	 * @param m2RepositoryPath the path to the m2repository
+	 */
+	public MavenLauncher(String projectRoot, String m2RepositoryPath, boolean includeTest) {
+		this.m2RepositoryPath = m2RepositoryPath;
+
+		if (!new File(projectRoot).isDirectory()) {
+			throw new SpoonException(projectRoot + " has to be a folder");
+		}
+
+		InheritanceModel model;
+		try {
+			model = readPOM(projectRoot, null);
+		} catch (Exception e) {
+			throw new SpoonException("Unable to read the pom", e);
+		}
+		if (model == null) {
+			throw new SpoonException(" Unable to create the model, pom not found?");
+		}
+
+		// source
+		List<File> sourceDirectories = model.getSourceDirectories();
+		for (File sourceDirectory : sourceDirectories) {
+			this.addInputResource(sourceDirectory.getAbsolutePath());
+		}
+
+		// test
+		if (includeTest) {
+			List<File> testSourceDirectories = model.getTestDirectories();
+			for (File sourceDirectory : testSourceDirectories) {
+				this.addInputResource(sourceDirectory.getAbsolutePath());
+			}
+		}
+
+		// dependencies
+		List<File> dependencies = model.getDependencies();
+		String[] classpath = new String[dependencies.size()];
+		for (int i = 0; i < dependencies.size(); i++) {
+			File file = dependencies.get(i);
+			classpath[i] = file.getAbsolutePath();
+		}
+		this.getModelBuilder().setSourceClasspath(classpath);
+	}
+
+
+	private InheritanceModel readPOM(String path, InheritanceModel parent) throws IOException, XmlPullParserException {
+		File parentPOM = Paths.get(path, "pom.xml").toFile();
+		if (!parentPOM.exists()) {
```

Comment:
What about distant parentPom like in https://github.com/pac4j/pac4j/blob/master/pom.xml for example? I had some issues with that in my own code, can you check it works well here?

Reply:
No idea if it works, I will not spend more time on this PR. I don't need this feature.